Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticSystem.cs

```diff
@@ -0,0 +1,196 @@
+        /// </value>
+        [SerializeField]
+        public float minStretch;
+
+        /// <value>
+        /// Represents the upper bound of the extent,
+        /// specified as the norm of the n-dimensional extent
+        /// </value>
+        [SerializeField]
+        public float maxStretch;
+
+        /// <value>
+        /// Whether the system, when approaching the upper bound,
+        /// will treat the upper bound as a snap point.
+        /// </value>
+        [SerializeField]
+        public bool snapToMax;
+
+        /// <value>
+        /// Points inside the extent to which the system will snap.
+        /// </value>
+        [SerializeField]
+        public T[] snapPoints;
+    }
+
+    /// <summary>
+    /// Properties of the damped harmonic oscillator differential system.
+    /// </summary>
+    [Serializable]
+    public struct ElasticProperties
+    {
+        /// <value>
+        /// Mass of the simulated oscillator element
+        /// </value>
+        [SerializeField]
+        public float mass;
+        /// <value>
+        /// Hand spring constant
+        /// </value>
+        [SerializeField]
+        public float hand_k;
+        /// <value>
+        /// End cap spring constant
+        /// </value>
+        [SerializeField]
+        public float end_k;
+        /// <value>
+        /// Snap point spring constant
+        /// </value>
+        [SerializeField]
+        public float snap_k;
+        /// <value>
+        /// Extent at which snap points begin forcing the spring.
+        /// </value>
+        [SerializeField]
+        public float snap_radius;
+        /// <value>
+        /// Drag/damper factor, proportional to velocity.
+        /// </value>
+        [SerializeField]
+        public float drag;
+    }
+
+    /// <summary>
+    /// Represents a damped harmonic oscillator over an
+    /// N-dimensional vector space, specified by generic type T.
+    /// 
+    /// This extensibility allows not just for 1, 2, and 3-D springs, but
+    /// allows for 4-dimensional quaternion springs.
+    /// </summary>
+    internal abstract class ElasticSystem<T>
+    {
+        protected ElasticExtentProperties<T> extentInfo;
+        protected ElasticProperties elasticProperties;
+
+        protected T currentValue;
+        protected T currentVelocity;
+
+        public ElasticSystem(T initialValue, T initialVelocity,
+                                ElasticExtentProperties<T> extentInfo, ElasticProperties elasticProperties)
+        {
+            this.extentInfo = extentInfo;
+            this.elasticProperties = elasticProperties;
+            currentValue = initialValue;
+            currentVelocity = initialVelocity;
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the forcing/desired value.
+        /// </summary>
+        /// <param name="forcingValue">Input value, for example, a desired manipulation position.</param>
```

Comment:
Can you add more details on what the forceValue is? In the current only caller this is the distance between hands. I agree that it doens't make sense to say "distanceBetweenHands" but there's still a gap between the naming of this and understability of what this really is.

In general when designing abstract functions, if we're going to be creating new types/concepts (i.e. forcingValue) it's very important to describe the properties/things around that.

Reply:
A "forcing" value is the typical name for the "input" to a damped harmonic oscillator; see https://en.wikipedia.org/wiki/Forcing_function_(differential_equations)

I can explain this further in the comments.